packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -100,12 +103,20 @@ if (!ExecutionEnvironment.canUseDOM) {
   let previousFrameTime = 33;
   let activeFrameTime = 33;
 
-  const frameDeadlineObject = {
-    didTimeout: false,
-    timeRemaining() {
-      const remaining = frameDeadline - now();
-      return remaining > 0 ? remaining : 0;
-    },
+  const buildFrameDeadlineObject = function(didTimeout: boolean) {
+    return {
+      didTimeout,
+      timeRemaining() {
+        const remaining = frameDeadline - now();
+        return remaining > 0 ? remaining : 0;
+      },
+    };
+  };
+
+  // define a helper for this because they should usually happen together
+  const clearTimeoutTimeAndScheduledCallback = function() {
+    timeoutTime = -1;
+    scheduledRICCallback = null;
```

Comment:
The name of this function and its implementation are identical. I think that's a smell. Seems like the point of the helper is so you don't accidentally reset the callback without also resetting `timeoutTime`. So maybe can we give it a more semantic name, like `resetScheduledCallback`?